pkg/controller/job/job_controller.go

```diff
@@ -1456,6 +1508,13 @@ func (jm *Controller) manageJob(ctx context.Context, job *batch.Job, jobCtx *syn
 		var indexesToAdd []int
 		if isIndexedJob(job) {
 			indexesToAdd = firstPendingIndexes(jobCtx, int(diff), int(*job.Spec.Completions))
+			if hasBackoffLimitPerIndex(job) {
+				indexesToAdd, remainingTime = jm.getPodCreationInfoForIndependentIndexes(logger, indexesToAdd, jobCtx.podsWithDelayedDeletionPerIndex)
```

Comment:
I just noticed that the backoff wouldn't increase if all failed pods match an `Ignore` rule, which could lead to fast retries.

Perhaps we can consider keeping a second counter for absolute number of failures in beta.

Related issue for regular backoffLimit: https://github.com/kubernetes/kubernetes/issues/119378

Reply:
I see, I think I could probably fix it tomorrow. The second counter could also be optional just to count the ignored failures.

Reply:
I think instead of keeping the absolute number explicit I prefer to have another counter for ignored failures, but there are two possibilities I see:
1. extend the semantics of the annotation to allow for second integer after comma
2. introduce a new annotation such as `job-index-failure-ignored-count`

I started with (1.) so that no API changes are needed. However, this wasn't true since still some API documentation for the API field had to change, so I'm now leaning towards (2.) as cleaner. PTAL. FYI @deads2k.